pkg/controller/servicebindingrequest/retriever.go

```diff
@@ -28,6 +28,24 @@ const (
 	secretPrefix  = "urn:alm:descriptor:servicebindingrequest:env:object:secret"
 )
 
+func getNestedValue(key string, sectionMap interface{}) (string, error) {
```

Comment:
I would expect to have `getNestedValue` as parte of the `Retriever` type. Is there a reason for standalone func?

Reply:
Retriever object is not accessed anywhere within the function. So, the standalone function would be fine.

Reply:
Indeed it does not access, but it's also not used anywhere else. Furthermore, is not ready to be used anywhere else, since it belongs to `Retriever` type. Since it does belong to Retriever and is only ready to be used in that context, I think it either deserves a standalone type, or join Retriever.
But leaving a global method with local scope is not something we should do in this case.

Reply:
By the way, we are also missing a doc-comment on this function, please.

Reply:
Done. 991bb922eb12a0d1658ecc69521e64b51d12bbfc

Reply:
Thanks :-)